### PR TITLE
Allow setting working directory for exec.Run

### LINKED ISSUE #536: Allow setting working directory for exec.Run
`exec.Run` does not seem to have a way to change the working directory atm.
Should be simple enough to implement if we add a field that will map to `exec.Cmd.Dir` in Go.

### PR DESCRIPTION
**Pull request by chai2010**
Fixes #536

### CODE REVIEW

File: pkg/tool/exec/exec.cue [hunk 1]
```diff
@@ -21,6 +21,10 @@ Run: {
 	// cmd is the command to run.
 	cmd: string | [string, ...string]
 
+	// dir specifies the working directory of the command.
+	// If dir is the empty string, runs the command in the current directory.
+	dir: *null | string
```
Comment (roman-mazur) [inline pkg/tool/exec/exec.cue:26]: I wonder if we really need the `null` value. The working directory is always some path, and the default value of the current directory could be expressed as `"."`.

**Review by roman-mazur — COMMENTED** [review]
I was going to submit a PR too, but you beat me on it :)

We should also update the test, checking that the directory is correctly mapped, similar to how it's done with env vars: https://github.com/cuelang/cue/blob/master/pkg/tool/exec/exec_test.go#L66

File: pkg/tool/exec/exec.cue (hunk 1 above)
Comment (mpvl) [reply pkg/tool/exec/exec.cue:26]: I'm fine with using `null`, although using `"."` would simplify the code below.

File: pkg/tool/exec/exec.go [hunk 2]
```diff
@@ -139,6 +139,15 @@ func mkCommand(ctx *task.Context) (c *exec.Cmd, doc string, err error) {
 
 	cmd := exec.CommandContext(ctx.Context, bin, args...)
 
+	if v := ctx.Obj.Lookup("dir"); v.Exists() {
```
Comment (mpvl) [inline pkg/tool/exec/exec.go:142]: Is this correct? This should also check if the value is null.

Would be good to test this in a `cmd/cue/cmd/testdata/*.txt` file.

Comment (chai2010) [reply pkg/tool/exec/exec.go:142]: PTAL

Comment (mpvl) [reply pkg/tool/exec/exec.go:142]: The test doesn't check the `null` case. 

Actually, looking more closely at the surrounding code, this would suffice and be more in style:

```
cmd.Dir, _ = ctx.Obj.Lookup("dir").String()
```
this will ignore any error by just setting the empty directory. 

This relies on the template definition to be type checked against the input. So any other value would be ignored.

File: pkg/tool/exec/exec.cue (hunk 1 above)
Comment (mpvl) [reply pkg/tool/exec/exec.cue:26]: An alternative here could simply be to use *"" as the default. For the stdout, for instance, `null` has some meaning over `""`. This is not the case here. I would remove `null` and either have no default or `*""` so that we later can use it to mean something else, if we ever need it.

**Review by mpvl — CHANGES_REQUESTED** [review]
Sorry for the delay. Took a bit closer look this time.

Almost done. No need for a test for `null` (see comments below).
